loader/common/log.c

```diff
@@ -13,6 +13,64 @@ enum log_level logger_set_level(enum log_level level)
     return prev;
 }
 
+#ifdef HYPER_SERIAL_LOG
+static inline void out8(u16 port, u8 data)
+{
+    asm volatile("outb %0, %1" ::"a"(data), "Nd"(port));
+}
+
+static inline u8 in8(u16 port) {
+    u8 out_value = 0;
+    asm volatile("inb %1, %0" : "=a"(out_value) : "Nd"(port) :);
+    return out_value;
+}
+#endif
+
+void serial_init(void) {
+#ifdef HYPER_SERIAL_LOG
+    const u16 com1 = 0x3F8;
+	const u16 baud_divisor = 12; // 115200 / 9600;
+
+    // Enable and set baud
+	out8(com1 + 3, 0b10000000 | in8(com1 + 3));
+	out8(com1 + 0, (baud_divisor << 8) >> 8);
+	out8(com1 + 1, baud_divisor >> 8);
+	out8(com1 + 3, ~0b10000000 & in8(com1 + 3));
+
+    // Set data width to 8
+    out8(com1 + 3, 0b00000011 | in8(com1 + 3));
+
+    // Stop bit 1
+    out8(com1 + 3, ~0b00000100 & in8(com1 + 3));
+
+    // No parity
+    out8(com1 + 3, ~0b00011100 & in8(com1 + 3));
+
+    // No interrupts
+	out8(com1 + 1, ~0b00001111 & in8(com1 + 1));
```

Comment:
Can we please make these magic numbers actual macros with bits so that these comments (at least partially) are not needed anymore?